### PR TITLE
`ChatLlm` implementation that supports Langchain `ChatModel`

### PR DESCRIPTION
**Pull request by mongodben**
Jira: n/a

## Changes

- [x] Add new ChatLlm constructor that supports the Langchain JS [ChatModel](https://js.langchain.com/docs/modules/model_io/chat/)
- [x] Test implementation with Azure OpenAI service + [FakeLlm](https://js.langchain.com/docs/integrations/chat/fake)
- [x] Make tool calling optional for a `ChatLlm`
- [x] Document new functionality
- [x] Updated some `scripts` service imports b/c was getting some `mongodb` package version incompatibility issues. 

## Notes

- This lets the chatbot server support all LLMs that Langchain supports 
- Working on this made me realize that our implementation of streaming is very based on the Azure OpenAI client's interface. We use few of the required fields. We should consider defining our own type for a streaming object that doesn't require all these unnecessary fields, but rather the subset that we do use.

### CODE REVIEW

File: packages/chatbot-server-mongodb-public/.env.example [hunk 1]
```diff
@@ -5,7 +5,7 @@ OPENAI_EMBEDDING_DEPLOYMENT=<deployment_name>
 OPENAI_EMBEDDING_MODEL=text-embedding-ada-002
 OPENAI_EMBEDDING_MODEL_VERSION=2023-03-15-preview
 OPENAI_CHAT_COMPLETION_DEPLOYMENT=<deployment_name>
-OPENAI_CHAT_COMPLETION_MODEL_VERSION=<model_version>
+OPENAI_CHAT_COMPLETION_MODEL_VERSION=2023-05-15
```
Comment (cbush) [inline packages/chatbot-server-mongodb-public/.env.example:8]: Intended?

File: packages/mongodb-chatbot-server/src/services/LangchainChatLlm.test.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  it("should generate response - streamed", async () => {
+    const langchainChatLlm = makeLangchainChatLlm({
+      chatModel: makeFakeChat(),
+    });
+    const res = await langchainChatLlm.answerQuestionStream({
+      messages,
+    });
+    let content = "";
+    for await (const event of res) {
+      // expect(event.choices[0].delta?.content).toBe(fakeResponses[0]);
```
Comment (cbush) [inline packages/mongodb-chatbot-server/src/services/LangchainChatLlm.test.ts:45]: Needed?

**Review by cbush — APPROVED** [review]
Couple things, looks good though!

File: packages/chatbot-server-mongodb-public/.env.example (hunk 1 above)
Comment (mongodben) [reply packages/chatbot-server-mongodb-public/.env.example:8]: yea

File: packages/mongodb-chatbot-server/src/services/LangchainChatLlm.test.ts (hunk 2 above)
Comment (mongodben) [reply packages/mongodb-chatbot-server/src/services/LangchainChatLlm.test.ts:45]: nop, let me remove

Comment (mongodben) [inline packages/mongodb-chatbot-server/src/services/LangchainChatLlm.test.ts:45]:
```suggestion
```

File: packages/chatbot-server-mongodb-public/.env.example (hunk 1 above)
Comment (mongodben) [reply packages/chatbot-server-mongodb-public/.env.example:8]: oh this should have been in mongodb-chatbot-server/.env.example, though can also be here. will add there too.

**Pull request merged**